### PR TITLE
Initial Merge: merging everything I have done so far.

### PR DESCRIPTION
**Pull request by ooMICAHoo**
Since we are hoping to establish a solidified workflow, I feel that I need to merge everything that I have worked on thus far. After this big merge, all future merges will only be for specific issues. Unfortunately, there is quite a bit that I am hoping to merge in. Here is a list of the changes I have made:
- sketching functionality improvements
- fixed delete point error
- fixed non manifold cut error
- comments

### CODE REVIEW

**Comment by ooMICAHoo** [comment]
I fixed the accidental code :) got rid of it.

File: op_polytrim/polytrim_datastructure.py [hunk 1]
```diff
@@ -464,25 +548,28 @@ def dist3d(v3):
             return delt.length
         
         closest_3d_point = min(self.pts, key = dist3d)
-        screen_dist = dist(loc3d_reg2D(context.region, context.space_data.region_3d, closest_3d_point))
+        point_screen_dist = dist(loc3d_reg2D(context.region, context.space_data.region_3d, closest_3d_point))
         
-        if screen_dist  < 20:
+        # If an input point is less than 20(some unit) away, stop and set hovered to the input point
```
Comment (vxlcoder) [inline op_polytrim/polytrim_datastructure.py]: units are pixels?

Comment (ooMICAHoo) [reply op_polytrim/polytrim_datastructure.py]: are they? I'm not sure..

Comment (patmo141) [reply op_polytrim/polytrim_datastructure.py]: yes they are pixels

**Pull request merged**